src/signal_generator_api.cpp

```diff
@@ -23,7 +23,7 @@ QList<int> SignalGenerator_API::getMode() const
 {
 	QList<int> list;
 
-	for (unsigned int i = 0; i < gen->channels.size(); i++) {
+	for (int i = 0; i < gen->channels.size(); i++) {
```

Comment:
Replacing all these IFs with c++ range-based for loops will make things easier. There will be no need to ask yourself what type "i" variable should have since it will be replaced with an iterator that usually is declared as auto. Also if container changes, let's say from qt containers to std containers, range-based for will not have to change, while in the actual form, "i" will need to be declared unsigned because that is what size() of std container returns.

Reply:
removed this commit for now;
will give this a deeper thought;

thanks
alex